src/main/kotlin/pt/up/fe/ni/website/backend/service/AuthService.kt

```diff
@@ -59,29 +59,43 @@ class AuthService(
 
     fun generateRecoveryToken(id: Long): String {
         val account = accountService.getAccountById(id)
-        return generateToken(account, Duration.ofMinutes(authConfigProperties.jwtRecoveryExpirationMinutes))
+        return generateToken(
+            account,
+            Duration.ofMinutes(authConfigProperties.jwtRecoveryExpirationMinutes),
+            usePasswordHash = true
+        )
     }
 
     fun getAuthenticatedAccount(): Account {
         val authentication = SecurityContextHolder.getContext().authentication
         return accountService.getAccountByEmail(authentication.name)
     }
 
-    private fun generateToken(account: Account, expiration: Duration, isRefresh: Boolean = false): String {
+    private fun generateToken(
+        account: Account,
+        expiration: Duration,
+        isRefresh: Boolean = false,
+        usePasswordHash: Boolean = false
+    ): String {
         val roles = if (isRefresh) emptyList() else getAuthorities() // TODO: Pass account to getAuthorities()
         val scope = roles
             .stream()
             .map(GrantedAuthority::getAuthority)
             .collect(Collectors.joining(" "))
         val currentInstant = Instant.now()
-        val claims = JwtClaimsSet
+        val claimsBuilder = JwtClaimsSet
             .builder()
             .issuer("self")
             .issuedAt(currentInstant)
             .expiresAt(currentInstant.plus(expiration))
             .subject(account.email)
             .claim("scope", scope)
-            .build()
+
+        if (usePasswordHash) {
+            claimsBuilder.claim("passwordHash", account.password)
+        }
+
+        val claims = claimsBuilder.build()
         return jwtEncoder.encode(JwtEncoderParameters.from(claims)).tokenValue
     }
```

Comment:
I don't know if this is what to do regarding the password serialization mentioned in this PR, but it seems right to me.

Reply:
Is this regarding password changing only 1 time with the link?

Reply:
Yes